src/adapters/summarize/summarize.ts

```diff
@@ -0,0 +1,191 @@
+      const { data: pubOrgEvents } = await octokit.activity.listPublicOrgEvents({
+        org: botConfig.git.org,
+        per_page: perPage,
+        page: currentPage,
+      });
+      console.log(pubOrgEvents);
+      pubOrgEvents.forEach((elem: any) => {
+        const elemTimestamp = new Date(elem.created_at as string).getTime();
+        if (elemTimestamp <= startTimestamp && elemTimestamp >= endTimestamp) {
+          //pass
+          elemList.push(elem);
+        } else if (elemTimestamp > startTimestamp) {
+          //outta range
+          //skip
+        } else {
+          //fail end
+          shouldFetch = false;
+        }
+      });
+      currentPage++;
+    } catch (error) {
+      shouldFetch = false;
+    }
+  }
+  JSONList = elemList;
+};
+
+const processEvents = () => {
+  let openedIssues: number = 0;
+  let closedIssues: number = 0;
+  let comments: number = 0;
+  let bountiesUSD: number = 0;
+  let openedPRs: number = 0;
+  let closedPRs: number = 0;
+  let mergedPRs: number = 0;
+  let commits: number = 0;
+  JSONList.forEach((elem: any) => {
+    const { type: eventType } = elem;
+    switch (eventType) {
+      case "IssuesEvent":
+        switch (elem.payload.action) {
+          case "opened":
+            openedIssues++;
+            break;
+          case "closed":
+            closedIssues++;
+            elem.payload.issue?.labels.forEach((elem: any) => {
+              if (elem.name.includes("Price")) {
+                const bountyUSD = parseInt(elem.name.toString().match(/[0-9]/gm).join(""));
+                bountiesUSD += bountyUSD;
+              }
+            });
+            break;
+          default:
+            break;
+        }
+        break;
+      case "IssueCommentEvent":
+        switch (elem.payload.action) {
+          case "created":
+            comments++;
+            break;
+          default:
+            break;
+        }
+        break;
+      case "PullRequestEvent":
+        switch (elem.payload.action) {
+          case "opened":
+            openedPRs++;
+            break;
+          case "closed":
+            if (elem.payload.pull_request?.merged === true) {
+              mergedPRs++;
+              commits += elem.payload.pull_request?.commits;
+            } else {
+              closedPRs++;
+            }
+            break;
+          default:
+            break;
+        }
+        break;
+      case "PushEvent":
+        commits += elem.payload.commits?.length;
+        break;
+      default:
+        break;
+    }
+  });
+  summaryInfo =
+    `<code>new issues: ${openedIssues}</code>\n` +
+    `<code>issues resolved: ${closedIssues}</code>\n` +
+    `<code>total user interactions count: ${comments}</code>\n` +
+    `<code>bounties given: ${bountiesUSD} USD</code>\n` +
+    `<code>new pulls: ${openedPRs}</code>\n` +
+    `<code>closed pulls: ${closedPRs}</code>\n` +
+    `<code>merged pulls: ${mergedPRs}</code>\n` +
+    `<code>total commits: ${commits}</code>\n`;
```

Comment:
This seems like a really solid summary implementation. You even are using whatthediff! Is this rendering an image and posting an image summary then?

Reply:
yes, it posts like this
![image](https://user-images.githubusercontent.com/109317559/218294978-46fec5e1-0ee5-4ad0-bbd1-01d2457d1a12.png)
@pavlovcik